diff --git a/highs/mip/HighsDomain.cpp b/highs/mip/HighsDomain.cpp
index 0c8064005c..80b8e08b1c 100644
--- a/highs/mip/HighsDomain.cpp
+++ b/highs/mip/HighsDomain.cpp
@@ -2154,6 +2154,7 @@ void HighsDomain::setDomainChangeStack(
     const std::vector<HighsInt>& branchingPositions) {
   infeasible_ = false;
   mipsolver->mipdata_->debugSolution.resetDomain(*this);
+  const bool noSymmetry = !mipsolver->mipdata_->detectSymmetries;
 
   if (!domchgstack_.empty()) {
     for (const HighsDomainChange& domchg : domchgstack_) {
@@ -2186,25 +2187,20 @@ void HighsDomain::setDomainChangeStack(
 
     if (k == stacksize) return;
 
-    // For redundant branching bound changes we need to be more careful due to
-    // symmetry handling. If these bound changes are redundant simply because
-    // the corresponding subtree was enumerated and hence the global bound
-    // updated, then we still need to keep their status as branching variables
-    // for computing correct stabilizers. They can, however, be safely dropped
-    // if they are either strictly redundant in the global domain, or if there
-    // is already a local bound change that makes the branching change
-    // redundant.
-    if (domchgstack[k].boundtype == HighsBoundType::kLower) {
-      if (domchgstack[k].boundval <= col_lower_[domchgstack[k].column]) {
-        if (domchgstack[k].boundval < col_lower_[domchgstack[k].column])
-          continue;
-        if (colLowerPos_[domchgstack[k].column] != -1) continue;
-      }
-    } else {
-      if (domchgstack[k].boundval >= col_upper_[domchgstack[k].column]) {
-        if (domchgstack[k].boundval > col_upper_[domchgstack[k].column])
-          continue;
-        if (colUpperPos_[domchgstack[k].column] != -1) continue;
+    // Remove redundant branching changes unless symmetry is active
+    if (noSymmetry) {
+      if (domchgstack[k].boundtype == HighsBoundType::kLower) {
+        if (domchgstack[k].boundval <= col_lower_[domchgstack[k].column]) {
+          if (domchgstack[k].boundval < col_lower_[domchgstack[k].column])
+            continue;
+          if (colLowerPos_[domchgstack[k].column] != -1) continue;
+        }
+      } else {
+        if (domchgstack[k].boundval >= col_upper_[domchgstack[k].column]) {
+          if (domchgstack[k].boundval > col_upper_[domchgstack[k].column])
+            continue;
+          if (colUpperPos_[domchgstack[k].column] != -1) continue;
+        }
       }
     }
 
diff --git a/highs/presolve/HPresolve.cpp b/highs/presolve/HPresolve.cpp
index 369d2daeff..6b862c28f7 100644
--- a/highs/presolve/HPresolve.cpp
+++ b/highs/presolve/HPresolve.cpp
@@ -9585,6 +9585,7 @@ HPresolve::Result HPresolve::detectParallelRowsAndCols(
       double rowScale = rowMax[parallelRowCand].first / rowMax[i].first;
       // check parallel case
       bool parallel = true;
+      storeRow(i);
       for (const HighsSliceNonzero& rowNz : getStoredRow()) {
         if (colsize[rowNz.index()] == 1)  // skip singletons
           continue;
@@ -9777,6 +9778,7 @@ HPresolve::Result HPresolve::equalityRowAddition(
                 static_cast<HighsCDouble>(scale) * model->row_upper_[stayrow]);
   addToRowUpper(removerow,
                 static_cast<HighsCDouble>(scale) * model->row_upper_[stayrow]);
+  reinsertEquation(removerow);
 
   // row is now a singleton row, doubleton equation, or a row
   // that contains only singletons and we let the normal row presolve
